.github/workflows/main.yml

```diff
@@ -40,25 +64,242 @@ jobs:
+      HEALTHCHECK_URL: ${{ needs.deploy.outputs.health_url }}
+      POSTMAN_API_KEY: ${{ secrets.POSTMAN_API_KEY }}
+      POSTMAN_WORKSPACE_ID: ${{ secrets.POSTMAN_WORKSPACE_ID }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Ensure prerequisites
+        run: |
+          command -v jq >/dev/null || (echo "jq not found" && exit 1)
+          test -n "$POSTMAN_API_KEY" || (echo "POSTMAN_API_KEY not set" && exit 1)
+          test -n "$POSTMAN_WORKSPACE_ID" || (echo "POSTMAN_WORKSPACE_ID not set" && exit 1)
+
+      - name: Wait for environment to be healthy
+        shell: bash
+        run: |
+          if [ -z "$HEALTHCHECK_URL" ]; then
+            echo "HEALTHCHECK_URL not set"; exit 1
+          fi
+          echo "Waiting for $HEALTHCHECK_URL to be healthy..."
+          ATTEMPTS=100
+          SLEEP=30
+          for i in $(seq 1 $ATTEMPTS); do
+            STATUS=$(curl -s -o /dev/null -w "%{http_code}" "$HEALTHCHECK_URL") || true
+            if [ "$STATUS" = "200" ]; then
+              echo "Healthcheck passed with 200"
+              exit 0
+            fi
+            echo "Attempt $i/$ATTEMPTS: got $STATUS, retrying in ${SLEEP}s..."
+            sleep $SLEEP
+          done
+          echo "Environment did not become healthy in time"
+          exit 1
+
+      - name: Fetch Postman collections and environment from workspace
+        id: fetch-postman
+        shell: bash
+        env:
+          AUTH_HEADER: ${{ format('X-Api-Key: {0}', secrets.POSTMAN_API_KEY) }}
+        run: |
+          set -euo pipefail
+          mkdir -p postman
+
+          # 1) Discover all collection UIDs in the workspace
+          COLLECTIONS_JSON=$(curl -s -H "X-Api-Key: ${POSTMAN_API_KEY}" \
+            "https://api.getpostman.com/collections?workspace=${POSTMAN_WORKSPACE_ID}")
+
+          echo "$COLLECTIONS_JSON" | jq -e . >/dev/null
+
+          echo "$COLLECTIONS_JSON" | jq -r '.collections[].uid' > postman/collection_uids.txt
+
+          COUNT=$(wc -l < postman/collection_uids.txt | tr -d ' ')
+          if [ "$COUNT" = "0" ]; then
+            echo "No collections found in workspace ${POSTMAN_WORKSPACE_ID}"; exit 1
+          fi
+
+          # OPTIONAL: Filter collections by name pattern
+          # Example: only run collections whose names start with "API -"
+          # echo "$COLLECTIONS_JSON" | jq -r '.collections[] | select(.name|test("^API -")) | .uid' > postman/collection_uids.txt
+
+          # 2) Download each collection JSON
+          while read -r COLLECTION_UID; do
+            COL_JSON=$(curl -s -H "X-Api-Key: ${POSTMAN_API_KEY}" \
+              "https://api.getpostman.com/collections/${COLLECTION_UID}")
+            NAME=$(echo "$COL_JSON" | jq -r '.collection.info.name' | tr ' /' '__')
+            echo "$COL_JSON" > "postman/${NAME}.collection.json"
+            echo "Downloaded collection: ${NAME}"
+          done < postman/collection_uids.txt
```

Comment:
Why are we writing these to disk?

Is there not some memory space we can store these in for use across GHA steps? 🤔